lib/cake_pay/src/models/cake_pay_card.dart

```diff
@@ -59,16 +72,75 @@ class CakePayCard {
       cardImageUrl: json['card_image_url'] as String?,
       country: json['country'] as String?,
       fiatCurrency: fiatCurrency,
-      denominationsUsd: denominationsUsd,
-      denominations: denominations,
       minValueUsd: json['min_value_usd'] as String?,
       maxValueUsd: json['max_value_usd'] as String?,
       minValue: json['min_value'] as String?,
       maxValue: json['max_value'] as String?,
+      denominationItems: parsedDenomination,
     );
   }
 
+  List<Denomination> get denominationItemsUnique {
+    if (denominationItems.isEmpty) return const [];
+    final map = <String, Denomination>{};
+    for (final d in denominationItems) {
+      final key = d.value.toStringAsFixed(2);
+      if (!map.containsKey(key)) {
+        map[key] = d;
+      } else {
+        final current = map[key]!;
+        if (current.cardId != id && d.cardId == id) {
+          map[key] = d;
+        }
+      }
+    }
+    return map.values.toList();
+  }
+
+  bool get hasDenominations => denominationItemsUnique.isNotEmpty;
+
+  List<String> get denominations =>
+      denominationItemsUnique.map((e) => _fmtValue(e.value)).toList(growable: false);
+
+  static String _fmtValue(double v) {
+    final intPart = v.truncate();
+    if (v == intPart) return intPart.toString();
+    var s = v.toStringAsFixed(2);
+    s = s.replaceAll(RegExp(r'0+$'), '').replaceAll(RegExp(r'\.$'), '');
+    return s;
+  }
+
   static String stripHtmlIfNeeded(String text) {
     return text.replaceAll(RegExp(r'<[^>]*>|&[^;]+;'), ' ');
   }
 }
+
+class Denomination {
+  final double value;
+  final int? cardId;
```

Comment:
this should be required, and this is the one that you should send when creating an order if it's there, if not (in the case of a price range card i.e card that doesn't have denominations) then you send the card id in the card object